src/ameedf/jb8223/assignments/hw02/Clock.java

```diff
@@ -34,4 +34,106 @@
+    }
+
+    // addHours - with a single parameter which represents the number of hours to add.
+    public void addHours(int hoursToAdd) {
+        if (hoursToAdd >= minValueAllowed && hoursToAdd <= maxValueAllowed) {
+                this.hours = (this.hours + hoursToAdd) % 24;
+        }
+    }
+
+    // addMinutes - with a single parameter which represents the number of minutes to add.
+    public void addMinutes(int minutesToAdd) {
+        if (minutesToAdd >= minValueAllowed && minutesToAdd <= maxValueAllowed) {
+                addHours(minutesToAdd / 60);
+                this.minutes = (this.minutes + minutesToAdd) % 60;
+        }
+    }
+
+    // addSeconds - with a single parameter which represents the number of seconds to add.
+    public void addSeconds(int secondsToAdd) {
+        if (secondsToAdd >= minValueAllowed && secondsToAdd <= maxValueAllowed) {
+                addMinutes(secondsToAdd / 60);
+                this.seconds = (this.seconds + secondsToAdd) % 60;
+        }
+    }
+
+    // addMilliseconds - with a single parameter which represents the number of milliseconds to add.
+    public void addMilliseconds(int millisecsToAdd) {
+        if (millisecsToAdd >= minValueAllowed && millisecsToAdd <= maxValueAllowed) {
+                addSeconds(millisecsToAdd / 1000);
+                this.milliseconds = (this.milliseconds + millisecsToAdd) % 1000;
+        }
+    }
+
+    // print - with a single boolean parameter that indicates weather we want to display a 24 hours clock
+    public void print(boolean show24HoursFormat) {
+        if (show24HoursFormat) {
+            addZeroOrNotAndPrint();
+            System.out.printf("\n");
+        }
+        else {
+            int hoursValueKeeper = hours;
+            if (hours >= 12 && hours < 24) {
+                if (hours > 12)
+                    hours = hours % 12;
+                addZeroOrNotAndPrint();
+                System.out.println("PM");
+            }
+            else {
+                addZeroOrNotAndPrint();
+                System.out.println("AM");
+            }
+            hours = hoursValueKeeper;
+        }
+    }
+
+    // addZeroOrNotAndPrint - a helper method that does the actual time fraction printing
+    private void addZeroOrNotAndPrint() {
+        if (hours < 10)
+            System.out.printf("0" + hours + ":");
+        else
+            System.out.printf(hours + ":");
+
+        if (minutes < 10)
+            System.out.printf("0" + minutes + ":");
+        else
+            System.out.printf(minutes + ":");
+
+        if (seconds < 10)
+            System.out.printf("0" + seconds + ".");
+        else
+            System.out.printf(seconds + ".");
+
+        if (milliseconds < 10)
+            System.out.printf("00" + milliseconds + " ");
+        else if (milliseconds < 100)
+            System.out.printf("0" + milliseconds + " ");
+        else
+            System.out.printf(milliseconds + " ");
+    }
```

Comment:
1. try to use " { } " in if-else statements
2. it's better to build up one string and use the System.out.print/ln once

Reply:
Wanted to avoid one big, messy string.

Reply:
you didn't.. actually it's out there - between each 2 "System.out.print(..."
:)

Reply:
got your point, will revise.